### PR TITLE
docs: ✏️ Update JSDoc comments for procedures creating auth request

### PR DESCRIPTION
**Pull request by prashantasdeveloper**

### CODE REVIEW

File: src/AccountManagement.ts
Comment (polymath-eric) [inline src/AccountManagement.ts:115]:
```suggestion
   * @note this will create an AuthorizationRequest which has to be accepted by the target Account.
```

(nit) The field name is target so I think it's a bit clearer to use that word. OK to leave as is though.

**Review by polymath-eric — APPROVED** [review]
LGTM, please confirm if we really need the unused import before merging though.

**Review by monitz87 — CHANGES_REQUESTED** [review]
We should also add links to `AuthorizationRequest`, `Account` and `Identity` in all the relevant comments. The problem is that right now there is no way to disambiguate between classes/types/enummembers that have the same name, and typedoc is picking up the incorrect ones. A workaround is to use `[link description](path-in-markdown)` (i.e. `[Accounts](account.md)` or `[getPrimaryAccount](identity.md#getprimaryaccount)`) which is what should be output by typedoc if it was working as intended, but this defeats the purpose of having tsdoc compliant documentation. We could have the `prepareCodeForDocs` pre-process link tags and transform them to the appropriate markdown but I think that is a task in itself.

For now, I suggest we start by changing all the square bracket notated links (such as `[[Account | Accounts]]`) to `@link` tags, and add the same tags to the auth request notes that didn't have any

Also, you left out some methods with the `getReceived` link

File: src/AccountManagement.ts
Comment (prashantasdeveloper) [reply src/AccountManagement.ts:5]: The generated docs work fine but I liked using the link before generating the doc. For me it was easier to navigate to the respective link when I'm trying to understand the meaning of it (as a developer).

**Comment by polymath-eric** [comment]
I am alright with having the imports if they give us functionality, previously I thought they weren't giving us any benefit.

That said, its going to make sonar cloud complain + look confusing if someone doesn't know what they do.

My preference is not having imports tools think are unused, but its OK to leave them if you think the benefits are worth it.

File: src/AccountManagement.ts
Comment (monitz87) [inline src/AccountManagement.ts:113]:
```suggestion
   * @note this will create an {@link AuthorizationRequest | Authorization Request} which has to be accepted by the `targetAccount`.
```

File: src/AccountManagement.ts
Comment (monitz87) [inline src/AccountManagement.ts:132]:
```suggestion
   * @note this will create an {@link AuthorizationRequest | Authorization Request} which has to be accepted by the `beneficiary` Account.
```

File: src/Identities.ts
Comment (monitz87) [inline src/Identities.ts:39]:
```suggestion
   * @note this may create {@link AuthorizationRequest | Authorization Requests} which have to be accepted by the `targetAccount`.
```

File: src/api/entities/Asset/CorporateActions/index.ts
Comment (monitz87) [inline src/api/entities/Asset/CorporateActions/index.ts:78]:
```suggestion
   * @note this may create {@link AuthorizationRequest | AuthorizationRequests} which have to be accepted by the `target` Identity.
```

File: src/api/entities/Asset/Permissions.ts
Comment (monitz87) [inline src/api/entities/Asset/Permissions.ts:70]:
```suggestion
   * @note this will create an {@link AuthorizationRequest | Authorization Request} which has to be accepted by the `target` Identity.
```

File: src/api/entities/Asset/index.ts
Comment (monitz87) [inline src/api/entities/Asset/index.ts:186]:
```suggestion
   * @note this will create {@link AuthorizationRequest | Authorization Request} which has to be accepted by the `target` Identity.
```

File: src/api/entities/Asset/index.ts
Comment (monitz87) [inline src/api/entities/Asset/index.ts:437]:
```suggestion
   * @note this will create an {@link AuthorizationRequest | Authorization Request} which has to be accepted by the `target` Identity.
```

File: src/api/entities/Portfolio/index.ts
Comment (monitz87) [inline src/api/entities/Portfolio/index.ts:212]:
```suggestion
   * @note this will create an {@link AuthorizationRequest | Authorization Request} which has to be accepted by the `targetIdentity`.
```

File: src/api/entities/TickerReservation/index.ts
Comment (monitz87) [inline src/api/entities/TickerReservation/index.ts:189]:
```suggestion
   * @note this will create {@link AuthorizationRequest | Authorization Request} which has to be accepted by the `target` Identity.
```

File: src/api/procedures/setPermissionGroup.ts
Comment (monitz87) [inline src/api/procedures/setPermissionGroup.ts:45]:
```suggestion
 *   - An array of {@link TxGroup | Transaction Groups} that represent a set of permissions. A Custom Permission Group will be created with those permissions, and
```

**Review by monitz87 — CHANGES_REQUESTED** [review]
I know JSDoc supports not using the `|` to separate URL from text, but I'm not sure TSDoc does, and it's more readable this way anyway

File: scripts/prepareCodeForDocs.js
Comment (monitz87) [inline scripts/prepareCodeForDocs.js:81]:
```suggestion
   * @note this method is of type {@link ${kind}}, which means you can call {@link ${kind}.checkAuthorization | ${name}.checkAuthorization}
   *   on it to see whether the signing Account and Identity have the required roles and permissions to run it
```

File: src/AccountManagement.ts
Comment (monitz87) [inline src/AccountManagement.ts:124]:
```suggestion
   *   An {@link Account} or {@link Identity} can fetch its pending Authorization Requests by calling {@link Authorizations.getReceived | authorizations.getReceived}.
   *   Also, an Account or Identity can directly fetch the details of an Authorization Request by calling {@link Authorizations.getOne | authorizations.getOne}
```

Please change this in the others as well

**Review by monitz87 — CHANGES_REQUESTED** [review]
There is a linting error in `src/api/procedures/__tests__/leaveIdentity.ts`

**Pull request merged**
